src/Adapter/Carrier/CommandHandler/AddCarrierHandler.php

```diff
@@ -56,6 +58,10 @@ public function __construct(
      */
     public function handle(AddCarrierCommand $command): CarrierId
     {
+        if (!$command->getShopConstraint()->isEqual(ShopConstraint::allShops())) {
```

Comment:
Or:
```suggestion
        if (!$command->getShopConstraint()->forAllShops()) {
```
To avoid creating of new ShopConstraint.